src/auth/auth.service.ts

```diff
@@ -0,0 +1,42 @@
+import { HttpException, Injectable } from '@nestjs/common';
+import { UsersService } from '../users/users.service';
+import * as bcrypt from 'bcrypt';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private readonly userService: UsersService,
+    private readonly jwtService: JwtService,
+  ) {}
+
+  async signIn(credentials: { username: string; password: string }) {
+    const user = await this.userService.findByUsername(credentials.username);
+    if (!user) {
+        // User not found
+        throw new HttpException('Not found', 404);
+    }
+
+    const isPasswordValid = await bcrypt.compare(credentials.password, user.password);
+    if (!isPasswordValid) {
+        // Invalid credentials
+        throw new HttpException('Unauthorized', 401);
+    }
+
+    const payload = { username: user.username, sub: user.id };
+    const accessToken = this.jwtService.sign(payload, {
+      secret: process.env.JWT_SECRET,
+      expiresIn: '30min',
+    });
+
+    const refreshToken = this.jwtService.sign(payload, {
+      secret: process.env.JWT_REFRESH_SECRET,
```

Comment:
add environment variables `JWT_SECRET` and `JWT_REFRESH_SECRET` to `.env.example`